web/src/main/java/org/wso2/testgrid/web/api/TestPlanService.java

```diff
@@ -143,4 +150,43 @@ public Response createTestPlan(TestPlanRequest testPlanRequest) {
                             setDescription(e.getMessage()).build()).build();
         }
     }
+
+    /**
+     * Save the {@link TestPlanRequest} object as a YAML file.
+     * @param testPlanRequest TestPlanRequest needs to be saved.
+     * @throws TestGridException thrown when error occurred;
+     *              1.If file directory (which is not existing) can not be created.
+     *              2.If IOException occured while writing YAML file.
+     *              3.If the YAML file already exists.
+     */
+    private void persistAsYamlFile(TestPlanRequest testPlanRequest) throws TestGridException {
+        java.nio.file.Path path = Paths.get(Constants.TESTPLANS_DIR , testPlanRequest.getTestPlanName());
```

Comment:
Shall we import java.nio.file.Path package and use. Because it will make the code more readable.